package.json

```diff
@@ -45,27 +46,37 @@
     "**/__tests__/fixtures/"
   ],
   "private": true,
+  "lint-staged": {
+    "*.{js,jsx}": [
+      "eslint --ignore-path .gitignore --ignore-path .prettierignore --fix",
+      "git add"
+    ],
+    "*.{json,md,scss,yaml,yml}": [
+      "prettier --write",
+      "git add"
+    ]
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
```

Comment:
So this will prevent a user from committing things to their own branch? Is there a way to prevent husky from installing the hook? To me it seems like you should be able to commit whatever you want while working on something, then of course before merging and reviewing you have CI to catch this kind of stuff.

Reply:
In practise there is no harm..

Reply:
@octalmage this only catches style problems and prevents a lot of "fix style" commits by flagging things ahead of time. In most cases, style fixes can be fixed automatically, so it shouldn't add much friction.

Of course, you can also always use `--no-verify` when committing to bypass this altogether (which the check will tell you if something fails).